backend/curriculum_tracking/tests/test_activity_log_entry_creators.py

```diff
@@ -40,3 +84,102 @@ def test_that_only_one_entry_created_for_one_review(self):
         creators.log_topic_competence_review_done(review2)
         creators.log_topic_competence_review_done(review2)
         self.assertEqual(LogEntry.objects.count(), 2)
+
+    def test_two_entries_created_for_two_reviews_with_second_done_within_debounce_period_by_another_user(self):
+        review1 = TopicReviewFactory()
+        review2 = TopicReviewFactory(
+            status=review1.status,
+            timestamp=review1.timestamp,
+            comments=review1.comments,
+            topic_progress=review1.topic_progress,
+            reviewer_user=UserFactory()
+        )
+
+        creators.log_topic_competence_review_done(review1)
+        creators.log_topic_competence_review_done(review2)
+        self.assertEqual(LogEntry.objects.count(), 2)
+
+    def test_logging_the_same_entry_twice_only_produces_one_log_entry(self):
+        review1 = TopicReviewFactory()
+        creators.log_topic_competence_review_done(review1)
+        creators.log_topic_competence_review_done(review1)
+        self.assertEqual(LogEntry.objects.count(), 1)
+
+    def test_log_topic_creator_is_invoked_for_a_topic_review_else_attribute_error_is_raised(self):
+        review1 = TopicReviewFactory()
+        with self.assertRaises(AttributeError) as exception:
+            creators.log_project_competence_review_done(review1)
+        self.assertTrue(exception, AttributeError)
+        self.assertEqual(LogEntry.objects.count(), 0)
+
+
+class log_project_and_topic_competence_reviews_done_TESTS(APITestCase, APITestCaseMixin):
+    LIST_URL_NAME = "agilecard-list"
+    SUPPRESS_TEST_POST_TO_CREATE = True
+    FIELDS_THAT_CAN_BE_FALSEY = [
+        "code_review_competent_since_last_review_request",
+        "code_review_excellent_since_last_review_request",
+        "code_review_red_flag_since_last_review_request",
+        "code_review_ny_competent_since_last_review_request",
+        "requires_cards",
+        "required_by_cards",
+        "project_submission_type_nice",
+        "topic_needs_review",
+        "topic_progress",
+        "due_time",
+        "complete_time",
+        "review_request_time",
+        "start_time",
+        "tag_names",
+        "can_start",
+        "can_force_start",
+        "flavour_names",
+        "open_pr_count",
+        "oldest_open_pr_updated_time",
+        "users_that_reviewed_since_last_review_request"
+    ]
+
+    def verbose_instance_factory(self):
+        project = RecruitProjectFactory()
+        card = AgileCardFactory(recruit_project=project)
+        return card
+
+    def setUp(self):
+        self.api_url = self.get_list_url()
+
+    @mock.patch.object(curriculum_tracking.activity_log_entry_creators, 'log_topic_competence_review_done')
+    @mock.patch.object(curriculum_tracking.activity_log_entry_creators, 'log_project_competence_review_done')
```

Comment:
The same object is being mocked under different names, is this intentional?

Reply:
It's not the same function being mocked, the one is "log_topic_competence_review_done" and the other is "log_project_competence_review_done".  I mock both to assert that the correct one does get called and therefore the other does not get called depending on the scenario.

Reply:
These are still inappropriate calls since we never instantiate `activity_log_entry_creators` instanes ourselves. We do not treat that as an object.

Rather:

```
@mock.patch(curriculum_tracking.activity_log_entry_creators.log_topic_competence_review_done)
@mock.patch(curriculum_tracking.activity_log_entry_creators.log_project_competence_review_done)
```